crates/solver/src/settlement_rater.rs

```diff
@@ -221,3 +241,43 @@ impl SettlementRating for SettlementRater {
         ))
     }
 }
+
+async fn solver_balances(web3: &Web3, addresses: HashSet<H160>) -> Result<HashMap<H160, U256>> {
```

Comment:
I have a few concerns with this function:
1) `join_all` works with iterator so you don't have to `.collect()`
2) using array indexing is safe in this case but IMO should be replaced with iterators if possible
3) if one of the `.balance()` calls fails the whole call fails causing all the settlements to be discarded. Alternatively you could never return an error and simply set the balance to 0 if a call fails.


Small suggestions which would address all those concerns:
```rust
async fn solver_balances(web3: &Web3, addresses: HashSet<H160>) -> HashMap<H160, U256> {
    let futures = addresses.into_iter().map(|address| async move {
        (
            address,
            web3.eth().balance(address, None).await.unwrap_or_default(),
        )
    });

    std::iter::FromIterator::from_iter(futures::future::join_all(futures).await.into_iter())
}
```